src/common/Backend.ts

```diff
@@ -123,6 +128,7 @@ export class Backend {
     run(
         data: BackendRunRequest
     ): Promise<BackendSuccessResponse | BackendExceptionResponse | BackendAlreadyRunningResponse> {
+        this.abortController = new AbortController();
```

Comment:
Why does `run` create a new AC every time? Like, this won't even work correctly if you have concurrent `run`s, which is not ensured here. Also, if I understand AC correctly, then all requests after `abort` will fail until we call `run` again.

Why don't make it the responsibility of `abort` to create a new AC? It used up the current one, so it should replace it with a new one.

Reply:
> if you have concurrent runs

That's not even possible at the moment... But yeah, I'll move it to after the abort

Reply:
> That's not even possible at the moment...

There is nothing *here* that guarantees that. You're using global reasoning. Like, in general, assuming that the whole application works correctly for a small piece of code to work isn't exactly best practice...